pkg/uhttp/gocache.go

```diff
@@ -0,0 +1,170 @@
+package uhttp
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"net/http/httputil"
+	"time"
+
+	bigcache "github.com/allegro/bigcache/v3"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"go.uber.org/zap"
+)
+
+type GoCache struct {
+	ttl         time.Duration
+	rootLibrary *bigcache.BigCache
+}
+
+func NewGoCache(ctx context.Context, ttl int32, cacheMaxSize int, isLogLevel bool) (GoCache, error) {
```

Comment:
This should probably take a config struct instead of a bunch of args. That way we don't have to change the function definition if we add more configuration options in the future.

Reply:
Fixed